tools/debug_tools/src/lib.rs

```diff
@@ -0,0 +1,70 @@
+//! Debugging dev tools for Emergence.
+
+use bevy::{
+    asset::AssetServer,
+    diagnostic::{Diagnostics, FrameTimeDiagnosticsPlugin},
+    prelude::{
+        default, Color, Commands, Component, Plugin, Query, ReflectComponent, Res, Resource,
+        TextBundle, With,
+    },
+    reflect::Reflect,
+    text::{Text, TextSection, TextStyle},
+    time::Time,
+    ui::{PositionType, Style, UiRect, Val},
+};
+
+use bevy_console::*;
+
+use bevy_inspector_egui::{Inspectable, RegisterInspectable, WorldInspectorPlugin};
+use console::{print_to_log, PrintToLog};
+
+pub mod console;
+pub mod debug_ui;
+
+#[derive(Clone, Resource, Component, Copy, Debug, PartialEq, Eq)]
+pub struct DebugInfo {
+    /// Toggle global access to developer tools
+    pub dev_mode: bool,
+    /// Toggle developer console
+    pub enable_console: bool,
+    /// Toggle the debug tile labels
+    pub show_tile_label: bool,
+    /// Toggle render info
+    pub show_fps_info: bool,
+    /// Toggle displaying the egui inspector
+    pub show_inspector: bool,
+}
+
+impl Default for DebugInfo {
+    fn default() -> Self {
+        Self {
+            dev_mode: true,
+            enable_console: true,
+            show_tile_label: true,
+            show_fps_info: true,
+            show_inspector: true,
+        }
+    }
+}
+/// Generate Debug Tools plugin
+pub struct DebugToolsPlugin;
+
+/// Tells `bevy-inspector-egui` how to display the struct in the world inspector
```

Comment:
These two types are very unclear to me.

Reply:
They were added to allow for adding custom components to the inspector but that is not being used or planned for use atm so I removed them.